src/sentry/integrations/bitbucket/issues.py

```diff
@@ -119,7 +117,7 @@ def create_issue(self, data, **kwargs):
         try:
             issue = client.create_issue(data.get("repo"), data)
         except ApiError as e:
-            self.raise_error(e)
+            raise self.raise_error(e)
```

Comment:
Why this change?

Reply:
I believe this change was done to help with mypy's flow analysis, and it is probably safe to do assuming `raise_error` raises unconditionally, but I think it should not be done in this PR. Instead `raise_error` should return [`NoReturn`](https://mypy.readthedocs.io/en/stable/more_types.html#the-noreturn-type).

Reply:
`NoReturn` is pretty cool. I'll split this off into a separate PR. It'll be tricky to get working because `raise_error()` isn't implemented in the mixin. I think I can get around this with a shared interface.

Reply:
Ok I had to give up. I can't commit to these files at all without satisfying `pyright` and it's getting `reportUnboundVariable` errors here.

Reply:
Just tried it out locally because I couldn't believe pyright would let you down like that, but... yeah. I think this is fine then. It certainly beats `return raise_error()` which might accidentally noop.